MovieSaver/MVVM/ViewModel/Navigator.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  Navigation.swift
+//  MovieSaver
+//
+//  Created by Yushkevich Ilya on 10.04.22.
+//
+
+import Foundation
+import UIKit
+
+enum Destination {
+  case addMovie((Movie) -> Void)
+  case movieList
+  case movieDetail(Movie)
+  case baseChangeInfo(ChangeInfoViewControllerInputType, (ChangeInfoViewControllerOutputType) -> Void)
+}
+
+@MainActor
+final class Navigator {
+
+  // MARK: - Properties
+  // MARK: Public
+  static let instance = Navigator(navigationController: UINavigationController(rootViewController: MovieListViewController()))
+  // MARK: Private
+  private(set) weak var navigationController: UINavigationController?
+  
+  // MARK: - Lifecycle
+  private init(navigationController: UINavigationController) {
+    guard self.navigationController == nil else { return }
```

Comment:
`self.navigationController ` is always nil during initialization. You can omit this line